src/stream/binary_writer.rs

```diff
@@ -713,24 +712,23 @@ mod tests {
 
         let streaming_parser = BinaryReader::new(Cursor::new(buf_inner));
 
-        let events: Vec<Result<_, _>> = streaming_parser.collect();
```

Comment:
This collect into a `Vec` is intentional so that any errors that occur during reading are reported before any errors that occur during `Value::from_events`.

Reply:
Part of the second commit, so this will be dropped out of the PR.